File: app/cache.py
import asyncio
from dataclasses import dataclass, field


@dataclass
class M3UCache:
    upstream_etag: str | None = None
    filtered_etag: str | None = None
    filtered_body: bytes | None = None
    lock: asyncio.Lock = field(default_factory=asyncio.Lock)


@dataclass
class EPGCache:
    upstream_etag: str | None = None
    filtered_etag: str | None = None
    filtered_body: bytes | None = None
    last_fetch: float = 0.0
    lock: asyncio.Lock = field(default_factory=asyncio.Lock)

File: internal/cache/cache.go
package cache

import (
	"sync"
	"time"
)

type M3UCache struct {
	UpstreamETag string
	FilteredETag string
	FilteredBody []byte
	Mu           sync.Mutex
}

type EPGCache struct {
	UpstreamETag string
	FilteredETag string
	FilteredBody []byte
	LastFetch    time.Time
	Mu           sync.Mutex
}
